Angelo datastreams bode

cbode opened the pull request:
Dashboard specific datastreams created.  Python code can be generalized to all datastreams, though it is not optimized.

Commit:
Geo has elevation as third value

cbode committed:
added seasonal aggregation script

Commit:
datastream_seasonal.py now runs without error

code works, but there are no controls to check if a value is already
inserted.

cbode committed:
added gitignore

cbode committed:
Delete odm.pw

Commit:
Merge branch 'master' into angelo-datastreams-bode

Commit:
Finished Function Seasonal

Reran angelo json.  Corrected

Commit:
Schema Changes Aggregation changed to Aggregate

All files with the filename or term referencing “Aggregation” were
changed to “Aggregate” to be more consistent with other vocabularies.

cbode committed:
Schema Update mediaSize and Derived

Station: mediaSize object defined in station.json as per Issue #12.
Datastream.json still had "derived" as required, even though it had been
removed.  TODO: remove derived from all examples.

cbode committed:
migration directory created

All code for generating datastreams from the Sensor Database is now in
the migration directory.  Code will require path changes to function.

Commit:
Updated migration code

Removed derived boolean from examples.  Added csv for blue oak ranch
datastreams. Adjusted the location of the database password.

cbode committed:
Schema and vocabulary updates

Issue #20 resolved by updating station.json url links.  Examples were
also updated.

cbode committed:
modified media_caption

cbode committed:
removed non-dashboard datastreams

Commit:
Cumulation Aggregation

Separate script now exists for accumulation on a daily basis.  For now
this is only for precipitation.  It could also be used for solar
radiation.

cbode committed:
JSON exports finished

Dashboard datastreams are now available for Blue Oak Ranch and Angelo.
Sequence for creation is:
1. datastream_csv2json.py
2. datastream_seasonal.py
3. datastream_daily_cumulative.py
4. datastream_seasonal_json.py
Note: bug caused all json to have attributes of first record.  fixed.

jscottsf commented:
In some datastreams, the datastream_id under the datapoints_config is string. It should be numeric.

jscottsf commented:
In a lot of datastreams, the field "attributes" is present but empty. Do we want to require that this object contain at least one field when present. Can this be done with JSON schema?

jscottsf commented:
I started a standard .gitignore that I use for all other repos. We may at some point want to merge/optimize the .gitignore here with the standard one I use. Just to avoid having so many flavors. Nothing actionable for now. Maybe we add this as a GitHub issue/enhancement or task on Trello.

cbode commented:
I vote to use your standard .gitignore.  The only relevant part of mine was the ignoring of ".pw" files.

Commit:
Added new vocabulary time aggregation

This controlled vocabulary describes the time steps that will commonly
be used for derived datasets that aggregate datastreams.

Commit:
New Vocab Interval, Bug fixes to JSON

Added ds-interval to vocabulary.  This is for derived datasets. It
defines the interval used to aggregate.

Issue #23 datastream id’s were string. Fixed to integer.   Attribute
heigh and depth were also string. Fixed to float.  ‘attributes’ if
empty were showing up anyway, now only show up if there are values.

cbode commented:
All issues have been addressed.  Request merge upon review.

jscottsf commented on examples/vocabularies/ds-interval.json:
```diff
@@ -1,8 +1,8 @@
   "created_at": "2017-01-31T09:30:00Z",
   "description": "TBD",
-  "label": "Time Aggregate",
+  "label": "Time Interval",
```
Label should be "Interval", not "Time Interval".

cbode committed:
Vocabulary Interval Corrected

Mislabeling fixed as per #23
